src/main/java/org/jenkinsci/plugins/benchmark/core/BenchmarkPublisher.java

```diff
@@ -234,7 +222,7 @@ public void perform(@Nonnull Run<?, ?> run, @Nonnull FilePath filePath, @Nonnull
         }
         if (failed) {
             taskListener.getLogger().println(Messages.BenchmarkPublisher_CollectionSuccessButValidationFailure());
-            run.setResult(Result.FAILURE);
+            run.setResult(Result.UNSTABLE);
```

Comment:
Note this change. This is somewhat subjective, if you want I can remove and/or split this change into separate commit/PR.

With this the plugin behaves more like other testing plugins, giving the build a yellow icon when the check failed.

Reply:
Saw that change, no issue with it. My only concern is the adding of actions inside the perform sections.

Reply:
Going ahead with the merge.